spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/AppEngineJdbcInfoProviderAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.sql;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.gcp.autoconfigure.core.AppEngineCondition;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Provides the {@link CloudSqlJdbcInfoProvider} for apps running on Google App Engine.
+ *
+ * @author João André Martins
+ */
+@Configuration
+@ConditionalOnClass({com.google.cloud.sql.mysql.SocketFactory.class})
```

Comment:
Why is there such a condition there? Looks like an oversight. Or is MySQL specific to GAE?

Reply:
[App Engine also requires the socket factory to be in the dependencies](https://cloud.google.com/appengine/docs/standard/java/cloud-sql/using-cloud-sql-mysql).

Reply:
This is the exact signal from the classpath that CloudSQL-MySQL is intended to be used rather than the regular MySQL.

Reply:
I don't think that answers my question. Why is it MySQL specific? 

(please don't assume I know those features, I don't. Perhaps a document that describes the features will help me but something called `AppEngineJdbcInfoProviderAutoConfiguration` that is MySQL specific isn't right unless AppEngine only supports MySQL.

Reply:
I'm also a bit confused by the Cloud SQL story on App Engine Standard. One thing to keep in mind is that historically App Engine had its own special network path to many of the GCP services, including Cloud SQL. However, I believe that's changing now with the latest version of App Engine Standard for Java 8. I'll try to get clarity on this.

In any case, the way this class is set up is assuming that App Engine only supports the MySQL variant of Cloud SQL. So, the class should probably be renamed to `AppEngineCloudSqlMySqlJdbcInfoProviderAutoConfiguration`. :smile: 

cc: @saturnism @jabubake

Reply:
From https://cloud.google.com/appengine/docs/standard/java/cloud-sql/using-cloud-sql-postgres, it seems that connecting to PostgreSQL from inside App Engine is exactly the same as outside, in terms of the JDBC URL and driver class name.

Same thing doesn't happen with MySQL, where both URL and driver class names are special: https://cloud.google.com/appengine/docs/standard/java/cloud-sql/using-cloud-sql-mysql

Why is MySQL specific, I also don't know. :(